src/overlays/actors/ovl_En_Hata/z_en_hata.c

```diff
@@ -27,15 +26,106 @@ const ActorInit En_Hata_InitVars = {
     (ActorFunc)EnHata_Update,
     (ActorFunc)EnHata_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Hata/EnHata_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Hata/EnHata_Destroy.s")
+static UNK_TYPE4 EnHata_UnusedData[] = { 0x0A000939, 0x20010000, 0x00000000, 0x00000000, 0x00000000,
+                                         0x00000080, 0x00000000, 0x00050100, 0x001000F6, 0x00000000,
+                                         0x00000000, 0x00000000, 0x00000000, 0xFF000000 };
+
+static Vec3f EnHata_Vec = { 0, 0, 0 };
+
+extern AnimationHeader D_06000444;
+extern SkeletonHeader D_06002FD0;
+extern u32 D_060000C0;
+
+void EnHata_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnHata* this = THIS;
+    SkelAnime* skelAnime;
+    void* collisionPtr;
+    f32 frameCount;
+
+    collisionPtr = NULL;
+    frameCount = (f32)SkelAnime_GetFrameCount(&D_06000444.genericHeader);
+    Actor_SetScale(&this->dyna.actor, 0.0133333336562f);
+    skelAnime = &this->skelAnime;
+    SkelAnime_Init(globalCtx, skelAnime, &D_06002FD0, &D_06000444, NULL, NULL, 0);
+    SkelAnime_ChangeAnim(skelAnime, &D_06000444, 1.0f, 0.0f, frameCount, 0, 0.0f);
+    DynaPolyInfo_SetActorMove(&this->dyna, DPM_UNK);
+    DynaPolyInfo_Alloc(&D_060000C0, &collisionPtr);
+    this->dyna.dynaPolyId =
+        DynaPolyInfo_RegisterActor(globalCtx, &globalCtx->colCtx.dyna, &this->dyna.actor, (u32)collisionPtr);
+    this->dyna.actor.unk_F8 = 500.0f;
+    this->dyna.actor.unk_FC = 550.0f;
+    this->dyna.actor.unk_F4 = 2200.0f;
+    this->invScale = 6;
+    this->maxStep = 1000;
+    this->minStep = 1;
+    this->unk_278 = (s16)(Math_Rand_ZeroOne() * 65535.0f);
+}
+
+void EnHata_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnHata* this = THIS;
+    SkelAnime_Free(&this->skelAnime, globalCtx);
+    DynaPolyInfo_Free(globalCtx, &globalCtx->colCtx.dyna, this->dyna.dynaPolyId);
+}
+
+void EnHata_Update(Actor* thisx, GlobalContext* globalCtx) {
+    GlobalContext* gblCtx;
```

Comment:
This seems incorrect. I would replace it either by a temp to `envCtx` or a `this` temp, which are likely to trigger the same codegen changes.

Reply:
yeah i global context temp is very strange, hope we can figure something out for this

Reply:
given that a globalCtx temp was also needed in skjneedle I think this is something that we can revisit at a later time if we figure out what is causing it. to be clear gamestabled you tried the ideas above and it didnt work?

Reply:
Yes, I tried those ideas and more, this was the only solution I was able to get.

Reply:
ok. fix the undefined syms conflict then this is good to go imo

Reply:
also remove the data and reloc .s files